api/app/models/project.rb

```diff
@@ -416,6 +417,16 @@ def trigger_creation_event
     Event.trigger(EventType[:project_created], self)
   end
 
+  def ensure_restricted_access_notice_content
```

Comment:
Since we're making sure we have easy access to the default strings from settings, we can use those rather than repeating hard-coded strings in code. Otherwise, if we ever want to mess with this wording, we have to update it in two places.

```ruby
def ensure_restricted_access_notice_content!
  return unless restricted_access?

  settings = Settings.instance
    
  self.restricted_access_heading = settings.default_restricted_access_heading unless restricted_access_heading?
  self.restricted_access_body = settings.default_restricted_access_body unless restricted_access_body?
end
```